controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -249,6 +249,9 @@ func (w *Workload) ForwardEtcdLeadership(ctx context.Context, machine *clusterv1
 	if leaderCandidate == nil {
 		return errors.New("leader candidate cannot be nil")
 	}
+	if leaderCandidate.Status.NodeRef == nil {
+		return errors.New("leader has no node reference")
+	}
```

Comment:
Do we have a test for this? It seems a like a race condition between scale up and scale down, where the newest node doesn't have a reference yet

Reply:
We have a test for it, but it was passing. Very Volkswagen. Trying to find out why.

Reply:
Test added